test/integration/core/staking-limits.integration.ts

```diff
@@ -0,0 +1,232 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+import { mine } from "@nomicfoundation/hardhat-network-helpers";
+
+import { ether } from "lib";
+import { getProtocolContext, ProtocolContext } from "lib/protocol";
+
+import { Snapshot } from "test/suite";
+
+import { Lido } from "../../../typechain-types";
+
+describe("Staking limits", () => {
+  let ctx: ProtocolContext;
+  let lido: Lido;
+  let snapshot: string;
+  let stranger: HardhatEthersSigner;
+  let voting: HardhatEthersSigner;
+
+  before(async () => {
+    ctx = await getProtocolContext();
+    lido = ctx.contracts.lido;
+    [stranger] = await ethers.getSigners();
+    voting = await ctx.getSigner("voting");
+
+    snapshot = await Snapshot.take();
+  });
+
+  after(async () => await Snapshot.restore(snapshot));
```

Comment:
Does not look like scenario tests, can use beforeEach/afterEach snapshots

Reply:
yes!